dm/run.py

```diff
@@ -0,0 +1,180 @@
+                        altitude=l[3].strip())
+                  for l in lines]
+
+    p = Point(coords[0][1])
+    u = utm.from_latlon(p.y, p.x)
+    utm_zone = u[2]
+    hemisphere = "north" if p.y > 0 else "south"
+
+    log.MM_INFO('UTM - %s' % utm_zone)
+    log.MM_INFO('Hemisphere - %s' % hemisphere)
+
+    return {'utm_zone': utm_zone, 'hemisphere': hemisphere}
+
+
+def create_sysutm_xml(utm_zone):
+    '''
+    Generate SysUTM.xml
+    :param utm_zone:
+    :return: write xml
+    '''
+    proj = '+proj=utm +zone={} +ellps=WGS84 +datum=WGS84 +units=m +no_defs'.format(utm_zone)
+    with open('SysUTM.xml', 'a') as xml:
+        xml.write('<SystemeCoord>\n')
+        xml.write('\t<BSC>\n')
+        xml.write('\t\t<TypeCoord>  eTC_Proj4   </TypeCoord>\n')
+        xml.write('\t\t<AuxR>   1   </AuxR>\n')
+        xml.write('\t\t<AuxR>   1   </AuxR>\n')
+        xml.write('\t\t<AuxR>   1   </AuxR>\n')
+        xml.write('\t\t<AuxStr> ' + proj + '  </AuxStr>\n')
+        xml.write('\t</BSC>\n')
+        xml.write('</SystemeCoord>\n')
+    xml.close()
+
+
+# RUN
+if __name__ == '__main__':
+
+    args = config.config()
+
+    log.MM_INFO('Initializing NodeMICMAC app - %s' % system.now())
+    log.MM_INFO(args)
+
+    projectDir = io.join_paths(args.project_path, args.name)
+    imageDir = io.join_paths(projectDir, 'images')
+
+    IN_DOCKER = os.environ.get('DEBIAN_FRONTEND', False)
+
+    if IN_DOCKER:
+        mm3d = 'mm3d'
+    else:
+        mm3d = '/home/mm-aug2018/bin/mm3d' # for dev: locally installed micmac branch
+
+    try:
+        log.MM_INFO('Starting..')
+        os.chdir(imageDir)
+
+        image_ext = get_image_type()
+        projection = get_projection(image_ext)
+
+        log.MM_INFO(image_ext)
+        log.MM_INFO(projection)
+
+        create_sysutm_xml(projection['utm_zone'])
+
+        # generate xif to gps and xif to xml
+        kwargs_gps2txt = {
+            'ext': image_ext,
+            'mm3d': mm3d
+        }
+        system.run('{mm3d} XifGps2Txt .*.{ext}'.format(**kwargs_gps2txt))
+        system.run('{mm3d} XifGps2Xml .*.{ext} RAWGNSS'.format(**kwargs_gps2txt))
+
+        # generate image pairs based on match distance or auto
+        kwargs_ori = {
+            'distance': args.matcher_distance,
+            'mm3d': mm3d
+        }
+        if args.matcher_distance:
+            log.MM_INFO('Image pairs by distance: {}'.format(args.matcher_distance))
+            system.run('{mm3d} OriConvert "#F=N X Y Z" GpsCoordinatesFromExif.txt RAWGNSS_N ChSys=DegreeWGS84@RTLFromExif.xml MTD1=1 NameCple=dronemapperPair.xml DN={distance}'.format(**kwargs_ori))
+        else:
+            log.MM_INFO('Image pairs by auto-distance')
+            system.run('{mm3d} OriConvert "#F=N X Y Z" GpsCoordinatesFromExif.txt RAWGNSS_N ChSys=DegreeWGS84@RTLFromExif.xml MTD1=1 NameCple=dronemapperPair.xml DN='.format(**kwargs_ori))
+
+        # tie-points (SIFT default [patented] - not for commercial use)
```

Comment:
The SIFT patent has expired, so this can probably be used just fine in all settings. https://piero.dev/2019/04/the-sift-patent-has-expired/

Reply:
Thanks for that information. I didn't know this. Much appreciated!